### DISCUSSION TITLE
nested properties in request object

### DISCUSSION THREAD

**Discussion by camiant** [spatie/laravel-data#23] [Q&A]
How to automatically cast values when they are, for example, properties of an object inside your payload?

```
use Spatie\LaravelData\Data;

class SongData extends Data
{
    public function __construct(
        public string $title,
        public string $artist,
    ) {
    }
}
```

let's say in my example `title` and `artist` are properties of `song` object.. readable by laravel as `request('song.title')` and `request('song.artist')`

If the autobinding is not possible, what's the suggested workaround?

**Comment by rubenvanassche** [comment]
Magic creation functions should solve this problem: https://spatie.be/docs/laravel-data/v1/as-a-data-transfer-object/creating-a-data-object#magical-creation. You can just type a request as the parameter and then set the properties correctly into the data object.

**Reply by camiant** [reply]
Thanks for your attention, Ruben! According to the docs, magical creation works only `fromModel` or `fromString`..
Any change to make it work when data comes ... `fromRequest` ? 😊

**Reply by rubenvanassche** [reply]
Normally adding the `fromRequest` method to the data class with a `Request` as parameter this should work. That's how it has been implemented, when a request comes in the `Request` is passed into the `from` method which will automatically select the most correct method to handle the request.
